diff -Naur menu-cache-1.0.0/menu-cache-gen/menu-merge.c menu-cache-1.0.0.tpg/menu-cache-gen/menu-merge.c
--- menu-cache-1.0.0/menu-cache-gen/menu-merge.c	2014-10-23 13:09:24.000000000 +0000
+++ menu-cache-1.0.0.tpg/menu-cache-gen/menu-merge.c	2015-02-26 00:06:30.046321062 +0000
@@ -1355,7 +1355,7 @@
         g_free(merged);
         /* destroy all KDELegacyDirs */
         for (l = children; l; l = l->next)
-            if (fm_xml_file_item_get_tag(l->data) == menuTag_KDELegacyDirs)
+            if (l->data && fm_xml_file_item_get_tag(l->data) == menuTag_KDELegacyDirs)
             {
                 fm_xml_file_item_destroy(l->data);
                 l->data = NULL;
