diff -Naur menu-cache-1.0.0/menu-cache-gen/menu-compose.c menu-cache-1.0.0.tpg/menu-cache-gen/menu-compose.c
--- menu-cache-1.0.0/menu-cache-gen/menu-compose.c	2014-10-23 13:09:24.000000000 +0000
+++ menu-cache-1.0.0.tpg/menu-cache-gen/menu-compose.c	2015-02-26 00:08:30.035944612 +0000
@@ -1023,7 +1023,7 @@
     gboolean ok = FALSE;
 
     tmp = (char *)g_getenv("CACHE_GEN_VERSION");
-    if (sscanf(tmp, "%d.%u", &i, &req_version) == 2)
+    if (tmp && sscanf(tmp, "%d.%u", &i, &req_version) == 2)
     {
         if (i != VER_MAJOR) /* unsupported format requested */
             return FALSE;
